File: 4/1.js
const { sum } = require('../common.js')

const input = require('node:fs').readFileSync('input', { encoding: 'latin1' })
const lines = input.trim().split('\n')

const points = lines.map(line => {
	let [ , winners, numbers ] = line.match(/^[^:]+:([^|]+)\|(.+)$/)
	winners = winners.trim().split(/\s+/)
	numbers = numbers.trim().split(/\s+/)

	let count = 0
	for (const number of numbers) {
		if (winners.includes(number)) {
			if (count) count *= 2
			else count = 1
		}
	}

	return count
})

console.log(sum(points))

File: 24/1.py
from common import intersect, make_hailstone
from itertools import combinations

with open('input', encoding = 'latin1') as input:
	hailstones = [make_hailstone(line.strip()) for line in input]

count = 0
for c in combinations(hailstones, 2):
	try:
		p = intersect(*c)
	except ArithmeticError:
		continue

	for d in p:
		if (200_000_000_000_000 <= d <= 400_000_000_000_000):
			pass
		else:
			break
	else:
		count += 1

print(count)
